apps/astarte_trigger_engine/lib/astarte_trigger_engine/config.ex

```diff
@@ -21,32 +21,116 @@ defmodule Astarte.TriggerEngine.Config do
+  app_env :amqp_consumer_password, :astarte_trigger_engine, :amqp_consumer_password,
+    os_env: "TRIGGER_ENGINE_AMQP_CONSUMER_PASSWORD",
+    type: :binary,
+    default: "guest"
+
+  @envdoc "Virtual host for the AMQP consumer connection"
+  app_env :amqp_consumer_virtual_host, :astarte_trigger_engine, :amqp_consumer_virtual_host,
+    os_env: "TRIGGER_ENGINE_AMQP_CONSUMER_VIRTUAL_HOST",
+    type: :binary,
+    default: "/"
+
+  @envdoc "The name of the AMQP queue created by the events consumer"
+  app_env :amqp_events_queue_name, :astarte_trigger_engine, :amqp_events_queue_name,
+    os_env: "TRIGGER_ENGINE_AMQP_EVENTS_QUEUE_NAME",
+    type: :binary,
+    default: "astarte_events"
+
+  @envdoc "The name of the exchange on which events are published"
+  app_env :amqp_events_exchange_name, :astarte_trigger_engine, :amqp_events_exchange_name,
+    os_env: "TRIGGER_ENGINE_AMQP_EVENTS_EXCHANGE_NAME",
+    type: :binary,
+    default: "astarte_events"
+
+  @envdoc "The routing_key used to bind to TriggerEngine specific events"
+  app_env :amqp_events_routing_key, :astarte_trigger_engine, :amqp_events_routing_key,
+    os_env: "TRIGGER_ENGINE_AMQP_EVENTS_ROUTING_KEY",
+    type: :binary,
+    default: "trigger_engine"
+
+  @envdoc "A comma separated list of <host>:<port> values of accessible Cassandra nodes"
+  app_env :cassandra_nodes, :astarte_data_access, :cassandra_nodes,
+    os_env: "ASTARTE_CASSANDRA_NODES",
+    type: :binary,
+    default: "localhost:9042"
+
+  @envdoc "A list of host values of accessible Cassandra nodes formatted in the Xandra format"
+  app_env :xandra_nodes, :astarte_data_access, :xandra_nodes,
+    os_env: "ASTARTE_CASSANDRA_NODES",
+    type: XandraNode,
+    default: ["localhost"]
+
+  @envdoc "A list of {host, port} values of accessible Cassandra nodes in a cqex compliant format"
+  app_env :cqex_cassandra_nodes, :astarte_data_access, :cqex_cassandra_nodes,
+    os_env: "ASTARTE_CASSANDRA_NODES",
+    type: CassandraCqexNode,
+    default: [{"localhost", 9042}]
+
+  @envdoc "The port used by trigger engine"
+  app_env :port, :astarte_trigger_engine, :port,
+    os_env: "PORT",
+    type: :integer,
+    default: 4007
+
   @doc """
   Returns the AMQP events consumer connection options
   """
   def amqp_consumer_options do
-    Application.get_env(:astarte_trigger_engine, :amqp_consumer_options, [])
+    {:ok, host} = amqp_consumer_host()
+    {:ok, port} = amqp_consumer_port()
+    {:ok, username} = amqp_consumer_username()
+    {:ok, password} = amqp_consumer_password()
+    {:ok, virtual_host} = amqp_consumer_virtual_host()
+
+    {:ok,
+     [host: host, port: port, username: username, password: password, virtual_host: virtual_host]}
   end
 
   @doc """
   Returns the events name of the exchange on which events are published
   """
   def events_exchange_name do
-    Application.get_env(:astarte_trigger_engine, :amqp_events_exchange_name)
+    amqp_events_exchange_name()
```

Comment:
Instead of having two functions that do the same thing, either rename the env variable or change the calling code to call the new function (I'd go for the second one)